### PR TITLE
Positional-Only Parameters

### PR DESCRIPTION
**Pull request by mariocj89**
Proposal for positional only arguments as discussed on [python-dev](https://mail.python.org/pipermail/python-dev/2018-January/151794.html).

@pablogsal @larryhastings

### CODE REVIEW

**Comment by brettcannon** [comment]
FYI your build failed due to a reST error.

**Comment by mariocj89** [comment]
Thanks @brettcannon will fix it

**Comment by mariocj89** [comment]
Should be fixed now, Took some time to run the tool locally and found some other issues.

I had to split the authors as it seems I cannot put break lines in the middle of an author due to [this regex](https://github.com/python/peps/blob/0cfa95c9e9236e4e5e60cfb9a7ec630bae737866/pep0/pep.py#L244) I am happy to send a PR to change it if that is desired.

Thanks!

**Comment by Rosuav** [comment]
I have no objection to this becoming a PEP. IIRC it was almost going to just get implemented, but then the recommendation was to settle the details with a PEP, so the fact that it hasn't gone through -ideas isn't a problem.

**Comment by brettcannon** [comment]
One other editor vote is good enough for me. :) I've assigned a PEP number and will merge once the CI turns green.

File: pep-0570.rst [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+by hand.  But this results in a disconnect between the
+Python function signature and what it actually accepts,
+not to mention the work of implementing said argument parsing.
+
+==========
+Motivation
+==========
+
+The new syntax will allow developers to further control how their
+api can be consumed. It will allow restricting the usage of keyword
```
Comment (brettcannon) [inline pep-0570.rst]: Capitalize "API".

File: pep-0570.rst [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+    new_varargslist:
+      vfpdef (',' vfpdef)* ',' '/' [',' [varargslist]] | varargslist
+
+It will be added to the actual typedargslist and varargslist
+but for easier discussion is presented as new_typedargslist and new_varargslist
+
+
+=========================
+Possible implementations
```
Comment (brettcannon) [inline pep-0570.rst]: There's only one implementation strategy listed, so I think it's beyond "possible". 😉 Just call this "Implementation Plan" and drop the sub-section.

File: pep-0570.rst [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+----------------------------------
+
+This implementation will involve a full change of the Grammar. This will
+involve following the steps outlined in PEP 306 [#PEP306]_. In addition, other
+steps are needed including:
+
+* Modifying the code object and the function object to be aware of positional
+  only arguments.
+
+* Modifiying `ceval.c` (`PyEval_EvalCodeEx`, `PyEval_EvalFrameEx`...)
```
Comment (brettcannon) [inline pep-0570.rst]: You're using Markdown-style backticks instead of reST double backticks throughout the PEP.

File: pep-0570.rst [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+
+This does not intend to be a guide or a comprehensive recipe on how to implement
+this but a rough outline of the changes this will make to the codebase.
+
+The advantages of this implementation involve speed, consistency with the
+implementation of keyword-only parameters as in PEP 3102 and a simpler implementation
+of all the tools and modules that will be impacted by this change.
+
+============
+Alternatives
```
Comment (brettcannon) [inline pep-0570.rst]: This section is normally called "Rejected Ideas".

File: pep-0570.rst [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+============
+
+The following alternatives were discarded along this PEP
```
Comment (brettcannon) [inline pep-0570.rst]: You don't need this sentence.

File: pep-0570.rst [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+
+Always an option, just not adding it. It was considered
+though that the benefits of adding it is worth the complexity
+it adds to the language.
+
+---------------------
+After marker proposal
+---------------------
+
+A complain the approach has is the fact that the modifier of
```
Comment (brettcannon) [inline pep-0570.rst]: "complain the approach has" -> "complaint against the proposal is"

File: pep-0570.rst [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+  def (x, y, /, z):
+
+If we define that ``/`` makes only z position-only it won't be possible
+to call x and y via keyword argument. Finding a way to work around it
+will add confusion given that at the moment keyword arguments cannot be
+followed by positional arguments. ``/`` will therefore make both the
+preceding and following position-only.
+
+-------------------
+Per argument marker
```
Comment (brettcannon) [inline pep-0570.rst]: "Per-argument"

File: pep-0570.rst [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+Note the dot on arg1 and arg2. Even if this approach might look easier
+to read it has been discarded as ``/`` goes further inline with the
+keyword-only approach and is less error prone.
+
+
+----------------
+Using decorators
+----------------
+
+
```
Comment (brettcannon) [inline pep-0570.rst]: Extra newline

File: pep-0570.rst [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+It has been suggested in python-ideas [#python-ideas-decorator-based]_ to provide
```
Comment (brettcannon) [inline pep-0570.rst]: "on python-ideas"

**Comment by brettcannon** [comment]
Actually reviewed the PEP and found a couple of things to change.

**Comment by mariocj89** [comment]
Updated, thanks a lot for the review. Really appreciated :)

**Comment by mariocj89** [comment]
If you want to check the output html, it is available here:
https://mariocj89.github.io/peps/pep-0570

**Comment by brettcannon** [comment]
Thanks! It's now merged. For future changes, send a PR and we will merge it. Please keep in mind, though, that discussion of the PEP should not occur on this repo (e.g. people shouldn't file issues asking questions about the PEP).

**Pull request merged**